core/renderers/common/info.js

```diff
@@ -139,11 +139,10 @@ Blockly.blockRendering.RenderInfo = function(renderer, block) {
   this.rows = [];
 
   /**
-   * The total number of input rows added onto the block.
-   * @type {number}
-   * @protected
+   * An array of input rows on the block.
+   * @type {!Array.<!Blockly.blockRendering.InputRow>}
    */
-  this.inputRowNum_ = 1;
+  this.inputRows = [];
```

Comment:
Just curious why the change from just storing the number of inputs?

Reply:
I think that way we can update our "not so good looking" for loops from looping through all the rows to just the input rows.